processor/elasticinframetricsprocessor/processor_test.go

```diff
@@ -0,0 +1,88 @@
+package elasticinframetricsprocessor
+
+import (
+	"context"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"go.opentelemetry.io/collector/component"
+	"go.opentelemetry.io/collector/pdata/pcommon"
+	"go.opentelemetry.io/collector/pdata/pmetric"
+	"go.opentelemetry.io/collector/processor"
+	"go.uber.org/zap"
+)
+
+// mockRemapper is a mock implementation of the remapper interface for testing purposes.
+type mockRemapper struct {
+	called      bool
+	scopeMetric pmetric.ScopeMetrics
+	metricSlice pmetric.MetricSlice
+	resource    pcommon.Resource
+}
+
+func (m *mockRemapper) Remap(scopeMetric pmetric.ScopeMetrics, metricSlice pmetric.MetricSlice, resource pcommon.Resource) {
+	m.called = true
+	m.scopeMetric = scopeMetric
+	m.metricSlice = metricSlice
+	m.resource = resource
+}
+
+func TestProcessMetrics(t *testing.T) {
```

Comment:
What do you think adding a test case for the processor when created with newProcessor? In that case, we will be asserting the actual `opentelemetry-lib hostmetrics.NewRemapper` too.

Reply:
Done